Tella/Data/Database/TellaData.swift

```diff
@@ -316,19 +337,67 @@ extension TellaData {
             debugLog(error)
             return []
         }
-
+        
     }
-    func getUwaziTemplateById(id: Int) -> CollectedTemplate? {
+    func getUwaziTemplateById(id: Int?) -> CollectedTemplate? {
         do {
+            guard let id else { return nil }
             return try database.getUwaziTemplate(templateId: id)
         } catch let error {
             debugLog(error)
             return nil
         }
     }
-    func deleteAllUwaziTemplate(id: Int) {
+    func deleteUwaziTemplate(id: Int) -> Result<Bool,Error> {
         database.deleteUwaziTemplate(id: id)
     }
 }
 
+extension TellaData {
+    @discardableResult
+    func addUwaziEntityInstance(entityInstance:UwaziEntityInstance) -> Result<Int,Error>  {
+
+        if let instanceId = entityInstance.id {
+           let result = database.updateUwaziEntityInstance(entityInstance: entityInstance)
+            
+            if case .success = result {
+                self.shouldReloadUwaziInstances.send(true)
+                return .success(instanceId)
+            } else {
+                return .failure(RuntimeError(""))
+            }
+            
+        } else {
+            let result =  database.addUwaziEntityInstance(entityInstance: entityInstance)
+            self.shouldReloadUwaziInstances.send(true)
+            return result
```

Comment:
Maybe let's directly returne the error from the Result in the `updateUwaziEntityInstance` case for more clarity -> 

   ```
 func addUwaziEntityInstance(entityInstance: UwaziEntityInstance) -> Result<Int, Error> {
        if let instanceId = entityInstance.id {
            let result = database.updateUwaziEntityInstance(entityInstance: entityInstance)
            switch result {
            case .success:
                shouldReloadUwaziInstances.send(true)
                return .success(instanceId)
            case .failure(let error):
                return .failure(error)
            }
        } else {
            let result = database.addUwaziEntityInstance(entityInstance: entityInstance)
            shouldReloadUwaziInstances.send(true)
            return result
        }
    }
```